CleanSpec.mk

```diff
@@ -0,0 +1,2 @@
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/system/app/*)
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/system/priv-app/*)
```

Comment:
these clean rules seem too broad.

what is the use-case where they are needed?

if you already have an aosp build completed and then decide to add opengapps to your build, then I think you can just run `make installclean` and the system-app and system-priv-app will be cleaned up.

Reply:
The use-case: make an incremental build after adding repos of opengapps

Of course manually doing make installclean would solve that.
But the CleanStep.mk is faster and smarter. It just removes apps in /system instead of whole /system. Besides, people usually forget them need to do so (after adding opengapps) that results in a broken build. The CleanStep.mk automates it.

Note this is the standard way to handle such a problem in AOSP build system.
AOSP often adds new clean steps. For example, see

cd build/make; git log CleanSpec.mk